### ISSUE TITLE
onKernelResponse is not triggered in case of exception

### ISSUE THREAD

**Issue by olix21** [nelmio/NelmioCorsBundle#124]
Hello,

We are using NelmioCorsBundle in an API and if an exception is sent, Symfony does not trigger the `kernel.response` event but the `kernel.exception` one.

So the Access-Control-Allow-Origin is not set. It's a little bit annoying as if the http client send a wrong request, the request will blocked by the CORS instead of returning the reason.

Shouldn't we add `$this->dispatcher->addListener('kernel.exception', array($this, 'onKernelResponse'), 0);`?

**Comment by Seldaek** [comment]
Yes sounds like that might be correct, feel free to send a PR with a fix

**Comment by mayesto** [comment]
Hi @Seldaek, this is occurred when you throw some exception in event listener. When exception is thrown in controller everything is ok
